### ISSUE TITLE
Cannot get correct result in eval_coco OR eval_voc

### ISSUE THREAD

**Issue by hadhe145** [ZiweiWangTHU/BiDet#11]
Hello, I encountered a problem. 
During the training phase, due to the small GPU memory problem, I changed batch_size in train_bidet_ssd.py to 16 and num_worker to 4, and lr remained at the default 1e-3. 
However, during the training process, the loss output will be NaN around 30,000 iterations, but when I evaluate the model parameters saved before becoming NaN, the output AP and mAP are both around 0.0xx.
Moreover, when I reduce the default learning rate to 1e-4, although the model loss will not have the problem of nan, it cannot obtain the correct evaluation result. 
May I ask if I have unfinished configuration operations and how I will solve this problem.

**Comment by Wuziyi616** [comment]
Hi! Thank you for your attention to our work. Actually we discovered in our experiments that, large learning rate is very important for training binary detectors. And we guess it's because large lr can avoid local minima for the binary neural network. In addition, if you use large lr, you need to use a large batch size to stabilize the training of the network. This might be why your loss goes to NaN when using batch_size=16 and lr=1e-3.

I also tried lr=1e-4 with batch_size=32 before, indeed the training process is more stable, but the performance is bad. If I remember correctly, we achieved ~60% mAP on VOC using this lr scheduler (lr starts from 1e-4 and decay to 1e-5 and 1e-6 when the loss stops decreasing).

So, from my perspective, I recommend you to use multiple GPUs to train on batch_size=32, because large initial lr is very important if you want to get good performance. If you can't find a way to train on large batch_size, maybe you can use some tricks like lr warmup to stabilize the early stage of training. But I'm sure whether the mAP will be as good as reported in our paper.

**Comment by hadhe145** [comment]
Thank you very much for your explanation, I will try again!
